jni/src/test/java/com/horizen/certnative/NaiveThresholdSignatureWKeyRotationProofTest.java

```diff
@@ -0,0 +1,272 @@
+package com.horizen.certnative;
+
+import com.horizen.TestUtils;
+import com.horizen.librustsidechains.FieldElement;
+import com.horizen.schnorrnative.SchnorrKeyPair;
+import com.horizen.schnorrnative.SchnorrPublicKey;
+import com.horizen.schnorrnative.SchnorrSignature;
+import com.horizen.schnorrnative.SchnorrKeysSignaturesList;
+import com.horizen.provingsystemnative.ProvingSystem;
+import com.horizen.provingsystemnative.ProvingSystemType;
+import org.junit.BeforeClass;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+
+public class NaiveThresholdSignatureWKeyRotationProofTest {
+
+    static int keyCount = 4;
+    static int customFieldsNum = 1;
+    static long threshold = 4;
+    static int backwardTransferCount = 10;
+    static boolean zk = false;
+
+    static int prevEpochNumber = 9;
+    static int epochNumber = 10;
+    static long prevBtrFee = 100L;
+    static long btrFee = 100L;
+    static long prevFtMinAmount = 200L;
+    static long ftMinAmount = 200L;
+
+    static int maxProofPlusVkSize = 9 * 1024;
+
+    SchnorrKeysSignaturesList keysSignaturesList;
+    WithdrawalCertificate withdrawalCertificate;
+    WithdrawalCertificate prevWithdrawalCertificate;
+
+    FieldElement scId;
+    FieldElement prevScId;
+    FieldElement genesisKeyRootHash;
+    FieldElement endCumulativeScTxCommTreeRoot;
+    FieldElement prevEndCumulativeScTxCommTreeRoot;
+
+    static String snarkPkPathNoCustomFields = "./test_snark_pk";
+    static String snarkVkPathNoCustomFields = "./test_snark_vk";
+    static String snarkPkPathCustomFields = "./test_snark_pk_with_custom_fields";
+    static String snarkVkPathCustomFields = "./test_snark_vk_with_custom_fields";
```

Comment:
These names might create conflicts. Let's change the name into something like `...test_cert_keyrot_snark...`

Reply:
Also, in this instance, there necessary will be at least 1 custom field

Reply:
Done